python/singa/autograd.py

```diff
@@ -4698,6 +4698,82 @@ def cossim(a, b):
     return CosSim()(a, b)[0]
 
 
+class UpSample(Operator):
+    """
+    Init a UpSample operator
+    """
+
+    def __init__(self, mode, scales):
+        """
+        Args:
+            scales (list[int]): The scale array along each dimension. It takes 
+                value greater than or equal to 1. 
+        """
+        super(UpSample, self).__init__()
+        self.scales = scales
+        self.mode = mode
+        if self.mode != "nearest":
+            assert False, "only support nearest mode."
+
+    def forward(self, x):
+        """
+        forward of UpSample
+        Args:
+            x (CTensor): input tensor.
+        Returns:
+            the output CTensor.
+        """
+        if isinstance(self.scales, np.ndarray):
+            self.scales = self.scales.tolist()
+        else:
+            self.scales = list(self.scales)
+        self.x_shape = list(x.shape())
+        for axis, s in zip(range(len(self.scales)), self.scales):
+            s = int(s)
+            if s == 1:
+                continue
+            x = x.Repeat([s, ], axis)
+        return x
+
+    def backward(self, dy):
+        """
+        backward of UpSample
```

Comment:
this kind of docstring is trivial and thus is not necessary.
we do not need docstring for every function/class/variable if the name is self-explainable

Reply:
fixed